bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -151,6 +163,62 @@ protected void processWatchEvent(WatchEvent<?> event, Kind<?> kind, Path path) {
             } catch (IOException e) {
                 logger.warn("Could not process config file '{}': {}", path, e);
             }
+        } else if (kind == ENTRY_DELETE) {
+            // Detect if a service specific configuration file was removed. We want to
+            // notify the service in this case with an updated empty configuration.
+            File configFile = path.toFile();
+            if (configFile.isHidden() || configFile.isDirectory() || !configFile.getName().endsWith(".cfg")) {
+                return;
+            }
+
+            exclusivcePIDMap.setFileRemoved(configFile.getAbsolutePath());
+            processOrphanExclusivePIDs();
+            storeCurrentExclusivePIDList();
+        }
+    }
+
+    private void loadExclusivePIDList() {
+        try (FileReader reader = new FileReader(exclusivePIDStore)) {
+            exclusivcePIDMap = GSON.fromJson(reader, ExclusivePIDMap.class);
+            if (exclusivcePIDMap != null) {
+                exclusivcePIDMap.initializeProcessPIDMapping();
+            }
+        } catch (JsonSyntaxException | JsonIOException | IOException e) {
+            logger.error("Error loading exclusive pids from {}.", exclusivePIDStore.getAbsolutePath(), e);

```

Comment:
Don't we also end up here if the file does not exist (which is a perfectly valid case)?

Reply:
yes, I will separate the IOException and only log to debug.